src/pyconverter/xml2py/writer.py

```diff
@@ -270,115 +408,257 @@ def write_source(
+                str_before_def = re.findall(pat.before_def, python_method)[0]
+                output = re.findall(pat.get_imports, str_before_def)
+                if len(output) == 0:
+                    fid.write(f"{python_method}\n")
+                    fid.close()
+                else:
+                    fid.close()
+                    import_handler(file_path, python_method, output)
+
+            all_commands.append(command.name)
+
+    if check_structure_map:
+        for command_name in name_map.keys():
+            if command_name not in all_commands:
+                raise Exception(f"{command_name} is not in the structure map")
+
+    if check_files:
+        for module_name in package_structure.keys():
+            for class_name, _ in package_structure[module_name].items():
+                file_path = os.path.join(library_path, module_name, f"{class_name}.py")
+                try:
+                    subprocess.run(["python", str(file_path)])
+                except Exception as e:
+                    raise RuntimeError(
+                        f"Failed to execute '{python_method}' from '{file_path}'."
+                    ) from e
+
+    write_global__init__file(library_path)
+    write__init__file(library_path)
+
+    logging.info(f"Commands written to {library_path}")
 
     # copy package files to the package directory
-    copy_package(template_path, new_package_path, clean)
+    copy_template_package(template_path, new_package_path, clean)
     graph_path = get_paths(xml_doc_path)[0]
     shutil.copytree(graph_path, os.path.join(new_package_path, "doc", "source", "images"))
+    return package_structure
 
-    return cmd_path
 
-
-def write_docs(commands, cmd_map, package_path):
+def write_docs(package_path, package_structure=None, config_path="config.yaml"):
     """Output to the autogenerated ``package`` directory.
 
     Parameters
     ----------
-    commands : list[XMLCommand]
-        List of XML commands.
+    package_path : str
+        Path to the new package folder.
 
-    cmd_map : dict
-        Dictionary with this format: ``{"command_name": "python_name"}``.
+    package_structure :
+        Dictionary with the following format:
+        ``{'python_module_name': [{'python_class_name': python_names_list}]}``.
 
-    path : str
-        Path to the new package folder.
+    config_path : str, optional
+        Path to the configuration file. The default is ``config.yaml``.
 
     Returns
     -------
     str
         Path to the new document page.
 
     """
+    library_name = get_config_data_value(config_path, "library_name_structured")
+    if library_name[0] == "src":
+        library_name.pop(0)
+    library_name = ".".join(library_name)
 
     doc_package_path = os.path.join(package_path, "doc/source")
     if not os.path.isdir(doc_package_path):
         os.makedirs(doc_package_path)
 
+    doc_src_content = """
+API documentation
+==================
+
+.. toctree::
+   :maxdepth: 1
```

Comment:
Is this supposed to be here?

Reply:
If you're referring to the indentation, it is the right place otherwise the indentation is taken into account.